semantics/llvm-syntax.k

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2013-2014 K Team. All Rights Reserved.
+/*@
+
+This is the syntax of LLVM.
+
+This syntax requires a input LLVM program without any comments on it.
+*/
+module LLVM-SYNTAX
+
+    /*
+    Define rules for non-keyword tokens. This is currently just a bunch
+    of hacks. They don't cover many valid forms of tokens, and they also
```

Comment:
This is because LLVM group don't have a nice document or grammar about their syntax. What they have is a SDF document which is old. In the beginning of the document, they claimed that this document is "bunch of hacks" in defining LLVM variables or names, because there is some programs which the SDF grammar cannot parse and there is some wrong programs which can be parsed by this SDF grammar. 

I am constantly changing the definition of variables and names when I am defining other constructors, since I need to test whether or not the variables and names make sense when I define other constructors. I think I will have a clean version once I have the full syntax of LLVM. 

This is the SDF LLVM grammar:
http://www.opensource.apple.com/source/llvmCore/llvmCore-2118/utils/llvm.grm